declarative-cicd/example-workflow.yaml

```diff
@@ -0,0 +1,48 @@
+name: Deploy Decodable resources
+
+on:
+  push:
+    branches:
+      - main
+  # pull_request:
+  workflow_dispatch:
+
+jobs:
+  setup-and-apply:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+
+    - name: Download and unpack the Decodable CLI
+      run: |
+        curl -L https://releases.decodable.co/decodable-cli/linux/amd64/decodable-cli-linux-amd64-1.20.0.tar.gz -o /tmp/decodable-cli.tar.gz && \
+        tar -xzf /tmp/decodable-cli.tar.gz -C /tmp && \
+        sudo mv /tmp/decodable-cli-linux-amd64-1.20.0/bin/decodable /usr/local/bin && \
+        rm -r /tmp/decodable-cli.tar.gz /tmp/decodable-cli-linux-amd64-1.20.0
+
+    - name: Create Decodable CLI auth file
+      run: |
+        mkdir -p ~/.decodable
+        cat <<EOF > ~/.decodable/auth
+        version: 1.0.0
+        tokens:
+          default:
+            refresh_token: ${{ secrets.DECODABLE_REFRESH_TOKEN }}
+        EOF
+
+    - name: Create Decodable CLI config file
+      run: |
+        mkdir -p ~/.decodable
+        cat <<EOF > ~/.decodable/config
+        version: 1.0.0
+        active-profile: default
+
+        profiles:
+          default:
+            account: ${{ vars.DECODABLE_ACCOUNT_NAME }}
+        EOF
+
+    - name: Apply all Decodable resources
+      run: decodable apply $(find declarative-cicd -name '*.yaml')
```

Comment:
Just out of curiosity: is there any specific reason for splitting up the resources into multiple files and then use find to collect them, over having everything in a single resource file?

Reply:
I can imagine in real life people may well use files and folders as ways to organise projects and resources. It'd make git diff etc a lot easier, for one thing.